src/CommonUtils.jl

```diff
@@ -71,7 +72,7 @@ function _getManifoldFullOrPart(mkd::ManifoldKernelDensity, aspartial::Bool=true
 end
 
 function Statistics.mean(mkd::ManifoldKernelDensity, aspartial::Bool=true; kwargs...)
-  return mean(_getManifoldFullOrPart(mkd,aspartial), getPoints(mkd, aspartial); kwargs...)
+  return mean(_getManifoldFullOrPart(mkd,aspartial), getPoints(mkd, aspartial), GeodesicInterpolation(); kwargs...)
```

Comment:
GeodesicInterpolation is currently way faster but still has dynamic dispatch.

Reply:
`ExtrinsicEstimation` should usually be the fastest way to compute `mean`. It doesn't work well for some manifolds but for Lie groups it is worth trying out when you prefer speed over accuracy.

Reply:
Thanks, I'll give it a try. With all the performance enhancements you helped with so far we are seeing a major improvement already.

Reply:
Great! Feel free to report any further issues you think I could help with.

Reply:
`ExtrinsicEstimation` gives an error:
```
MethodError: no method matching get_embedding(::GroupManifold{ℝ, ProductManifold{ℝ, Tuple{TranslationGroup{ManifoldsBase.TypeParameter{Tuple{2}}, ℝ}, SpecialOrthogonal{ManifoldsBase.TypeParameter{Tuple{2}}}}}, Manifolds.SemidirectProductOperation{RotationAction{LeftAction, TranslationGroup{ManifoldsBase.TypeParameter{Tuple{2}}, ℝ}, SpecialOrthogonal{ManifoldsBase.TypeParameter{Tuple{2}}}}}})
```
Don't worry about it now though. I think there are other optimizations that will make a bigger difference.